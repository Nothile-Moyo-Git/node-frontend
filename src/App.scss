@use "./scss/partials/animations" as animations;
@use "./scss/partials/branding" as branding;
@use "./scss/partials/typography";

*{ 
  font-size: 16px;
}

html {
  background: branding.$backdrop_color_one;
}

body {
  height: 100vh;
}

#root{
  height: 100%;
  display: flex;
  flex-direction: column;
}

.app {
  text-align: center;
  animation: fade-in 0.2s linear;

  &__content{
    padding-top: 1.5rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
  }

  &__title{
    margin: 0.5rem 0;
  }

  &__text{
    margin: 0.5rem;
  }
}
