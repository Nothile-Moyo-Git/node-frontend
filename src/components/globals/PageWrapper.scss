@use "../../scss/partials/mixins" as mixins;
@use "../../scss/partials/branding" as branding;
@use "../../scss/partials/responsive" as responsive;
@use "../../scss/partials/animations" as animations;

@use "../../scss/partials/scrollbars";

main{
    animation: fade-in 0.2s linear;
    height: 100%;
    margin-top: 55px;
    overflow-y: auto;
    display: flex;
    justify-content: center;

    @include responsive.breakpoint(tablet){
        margin-top: 0;
    }
}

.page{
    background: branding.$backdrop_color_one;
}

.menu-open{

    @include responsive.breakpoint(tablet){
        &__logged-in{

            .menu{
                height: 137px;
            }
        }

        &__logged-out{

            ul{
                height: 105px;
            }
        }
    }
}

.menu-closed{

    @include responsive.breakpoint(tablet){
        min-height: 100vh;
        margin-top: 0px;
        padding-top: 0px; 
        nav{
            display: none;
        }
    }
}

.footer{
    position: fixed;
    bottom: 0.75rem;
    right: 0.75rem;
    border: 0;

    svg{
        @include mixins.state(){
            path{
                stroke: branding.$brand_color_six;
            }
        }
    }

    path{
        stroke : white;
        transition: animations.$standard_transition;
    }
}