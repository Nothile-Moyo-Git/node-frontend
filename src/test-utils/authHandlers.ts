// handlers.ts
import { graphql } from "msw";

export const handlers = [
  graphql.query("PostUserDetailsResponse", (req, res, ctx) => {
    return res(
      ctx.data({
        PostUserDetailsResponse: {
          sessionCreated: "2024-01-01",
          sessionExpires: "2024-12-31",
          success: true,
          user: {
            _id: "1",
            name: "Nothile Moyo",
            email: "nothile@example.com",
            password: "secret",
            confirmPassword: "secret",
            status: "active",
            posts: [],
          },
        },
      }),
    );
  }),
];
