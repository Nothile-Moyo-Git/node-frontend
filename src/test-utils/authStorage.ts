/**
 * Date Created: 05/08/2025
 *
 * Author: Nothile Moyo
 *
 * Description: This is the local storage handler file for my tests
 * It will handle anything that needs to be set in local storage, in this case, authentication
 *
 * All local storage items are deleted when the request is complete
 */

/**
 * @name setMockAuthStorage
 *
 * @description Mocks setting authentication data for a user
 *
 * @type () => void
 */
export const setMockAuthStorage = ({
  token = "mock-token",
  userId = "mock-user-id",
  expiresIn = new Date(Date.now() + 3600000).toISOString(),
} = {}) => {
  localStorage.setItem("token", token);
  localStorage.setItem("userId", userId);
  localStorage.setItem("expiresIn", expiresIn);
};

export const clearAuthStorage = () => {
  localStorage.removeItem("token");
  localStorage.removeItem("userId");
  localStorage.removeItem("expiresIn");
};
