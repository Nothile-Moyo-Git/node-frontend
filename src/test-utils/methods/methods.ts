// Mock our fetch functioality
export const createFetchResponse = (
  data: unknown,
  status = 200,
  ok = true,
): Response => {
  return {
    ok,
    status,
    json: async () => data,
  } as Response;
};
