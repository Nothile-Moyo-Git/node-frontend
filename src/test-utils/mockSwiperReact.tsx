/**
 * Date created : 01/10/2025
 * Author : Nothile Moyo
 *
 * Test routing file, this will create routes which can be used in Jest for testing
 * We use this to be able to render our components
 *
 */

import { ReactElement } from "react";

export const Swiper = (children: ReactElement) => (
  <div data-testid="swiper">{children}</div>
);

export const SwiperSlide = (children: ReactElement) => {
  <div data-testid="swiper-slide">{children}</div>;
};

export default {
  Swiper,
  SwiperSlide,
};
