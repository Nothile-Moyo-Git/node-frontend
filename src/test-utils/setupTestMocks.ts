/**
 * Date created: 30/07/2025
 *
 * Author: Nothile
 *
 * Methods test file, this file defines reusable methods and hooks which can be used across my tests
 * These methods are intended to be used to replicate the official functionality of the app
 *
 */

import { useState } from "react";
import { User } from "../@types";

// Mock useNavigate so it works with jest
export const mockUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockUsedNavigate,
}));

// This allows us to be able to mock hooks in Jest otherwise we use the actual hook which triggers errors
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: jest.fn(),
}));

// Cast since we're using a ts file and we want to manipulate this
const mockedUseState = useState as jest.Mock;

const setMockState = jest.fn();

// Default mock implementation of usestate
mockedUseState.mockImplementation((init) => [init, setMockState]);

// Utility function to update mocked isLoading in tests
export const setAppStateMock = (
  isLoading: boolean,
  loadingError: boolean,
  user: User | undefined,
  sessionExpiryDate: string | undefined,
  sessionCreationDate: string | undefined,
) => {
  // Call the state in the same order but ensure that our initial values aren't set multiple times
  let hasSetLoading = false;
  let hasSetUser = false;
  const hasSetExpiry = false;

  mockedUseState.mockImplementation((init) => {
    if (typeof init === "boolean" && !hasSetLoading) {
      hasSetLoading = true;
      return [isLoading, setMockState];
    }
    if (typeof init === "boolean") {
      return [loadingError, setMockState];
    }
    if (init === undefined && !hasSetUser) {
      hasSetUser = true;
      return [user, setMockState];
    }
    /* if (typeof init === "string" && !hasSetExpiry) {
      hasSetExpiry = true;
      return [sessionExpiryDate, setMockState];
    }
    if (typeof init === "string") {
      return [sessionCreationDate, setMockState];
    } */
    return [init, setMockState];
  });
};
