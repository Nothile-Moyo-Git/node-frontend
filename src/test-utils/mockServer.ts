import { setupServer } from "msw/node";
import { handlers } from "./authHandlers";

export const server = setupServer(...handlers);
