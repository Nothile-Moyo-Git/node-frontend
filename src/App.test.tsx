import { expect } from "@jest/globals";
import App from "./App";
import { act } from "react-dom/test-utils";
import { render, screen } from "@testing-library/react";

// Importing mocks to be used for testing
import "./mocks/methods.test";

// Setup mocks and environment
beforeEach(() => {});

// Cleanup mocks and environment
afterEach(() => {});

// Handle the main authentication of the app
it("Renders the app successfully", () => {
  act(() => {
    render(<App />);
  });

  // Check if the app component is rendered and we navigate to it successfully
  const appComponent = screen.getByTestId("testid-app-component");
  expect(appComponent).toBeDefined();
});

