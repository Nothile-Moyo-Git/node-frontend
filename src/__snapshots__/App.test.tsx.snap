// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Tests Renders the app successfully 1`] = `
<div
  class="app"
  data-testid="test-id-app-component"
>
  <div
    class="loadingWrapper"
    data-testid="test-id-loading-spinner"
  >
    <span
      class="loadingSpinner"
    />
    <p
      class="loadingText"
    >
      Loading, please wait...
    </p>
  </div>
</div>
`;
